src/carplay.cs

```diff
@@ -0,0 +1,720 @@
+	}
+
+	interface ICPMapTemplateDelegate { }
+
+	[NoWatch, NoTV, NoMac, iOS (12,0)]
+	[Protocol, Model (AutoGeneratedName = true)]
+	[BaseType (typeof (NSObject))]
+	interface CPMapTemplateDelegate {
+
+		[Export ("mapTemplate:shouldShowNotificationForManeuver:")]
+		bool ShouldShowNotificationForManeuver (CPMapTemplate mapTemplate, CPManeuver maneuver);
+
+		[Export ("mapTemplate:shouldUpdateNotificationForManeuver:withTravelEstimates:")]
+		bool ShouldUpdateNotificationForManeuver (CPMapTemplate mapTemplate, CPManeuver maneuver, CPTravelEstimates travelEstimates);
+
+		[Export ("mapTemplate:shouldShowNotificationForNavigationAlert:")]
+		bool ShouldShowNotificationForNavigationAlert (CPMapTemplate mapTemplate, CPNavigationAlert navigationAlert);
+
+		[Export ("mapTemplateDidShowPanningInterface:")]
+		void DidShowPanningInterface (CPMapTemplate mapTemplate);
+
+		[Export ("mapTemplateWillDismissPanningInterface:")]
+		void WillDismissPanningInterface (CPMapTemplate mapTemplate);
+
+		[Export ("mapTemplateDidDismissPanningInterface:")]
+		void DidDismissPanningInterface (CPMapTemplate mapTemplate);
+
+		[Export ("mapTemplate:panBeganWithDirection:")]
+		void PanBegan (CPMapTemplate mapTemplate, CPPanDirection direction);
+
+		[Export ("mapTemplate:panEndedWithDirection:")]
+		void PanEnded (CPMapTemplate mapTemplate, CPPanDirection direction);
+
+		[Export ("mapTemplate:panWithDirection:")]
+		void Pan (CPMapTemplate mapTemplate, CPPanDirection direction);
+
+		[Export ("mapTemplate:willShowNavigationAlert:")]
+		void WillShowNavigationAlert (CPMapTemplate mapTemplate, CPNavigationAlert navigationAlert);
+
+		[Export ("mapTemplate:didShowNavigationAlert:")]
+		void DidShowNavigationAlert (CPMapTemplate mapTemplate, CPNavigationAlert navigationAlert);
+
+		[Export ("mapTemplate:willDismissNavigationAlert:dismissalContext:")]
+		void WillDismissNavigationAlert (CPMapTemplate mapTemplate, CPNavigationAlert navigationAlert, CPNavigationAlertDismissalContext dismissalContext);
+
+		[Export ("mapTemplate:didDismissNavigationAlert:dismissalContext:")]
+		void DidDismissNavigationAlert (CPMapTemplate mapTemplate, CPNavigationAlert navigationAlert, CPNavigationAlertDismissalContext dismissalContext);
+
+		[Export ("mapTemplate:selectedPreviewForTrip:usingRouteChoice:")]
+		void SelectedPreview (CPMapTemplate mapTemplate, CPTrip trip, CPRouteChoice routeChoice);
+
+		[Export ("mapTemplate:startedTrip:usingRouteChoice:")]
+		void StartedTrip (CPMapTemplate mapTemplate, CPTrip trip, CPRouteChoice routeChoice);
+
+		[Export ("mapTemplateDidCancelNavigation:")]
+		void DidCancelNavigation (CPMapTemplate mapTemplate);
+	}
+
+	[NoWatch, NoTV, NoMac, iOS (12,0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface CPNavigationAlert : NSSecureCoding {
+
+		// Defined inside CPNavigationAlert.h
+		// static NSTimeInterval const CPNavigationAlertMinimumDuration = 10;
+		[Static]
+		[Wrap ("10D")]
+		double NavigationAlertMinimumDuration { get; }
```

Comment:
Maybe this should be a constant in the code behind instead?

```csharp
partial class CPNavigationAlert {
    public const double MinimumDuration = 10;
}
```

Reply:
good point, that's closer to the original definition